src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/XamarinAndroidApplicationProject.cs

```diff
@@ -41,6 +43,10 @@ public XamarinAndroidApplicationProject (string debugConfigurationName = "Debug"
 			: base (debugConfigurationName, releaseConfigurationName)
 		{
 			if (Builder.UseDotNet) {
+				if (EnableMarshalMethods.HasValue) {
```

Comment:
This code runs in the constructor. My understanding is that if you set a property using the following style of code. The property set will happen after the constructor code has completed.

```
proj = new XamarinAndroidApplicationProject {
    EnableMarshalMethods = enableMarshalMethods,
};
```

The code after this new code is for setting the defaults which need to be in the csproj. 
You should probably use code like this https://github.com/xamarin/xamarin-android/blob/06ea40b567ac10b17cb4d024f08854c9cc31587f/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/XamarinAndroidApplicationProject.cs#L114

Reply:
Good catch sir :)